.github/workflows/CI-pipeline.yml

```diff
@@ -44,18 +44,22 @@ jobs:
           arr=()
           SERVICES=( ${{ env.SERVICES }} )
           git fetch origin master
-          for i in "$(git diff  --name-only origin/master..HEAD)"
+          changes=($(git diff  --name-only origin/master))
+          for i in "${changes[@]}"
           do
-            if [[ $i == *"internal/"* ]] || [[ $i == *"proto/"* ]] || [[ $i == "go.mod" ]] || [[ $i == "go.sum" ]] && [[ $i != *".md"* ]]; then
+            if [[ "${i}" =~ .*\.md.* ]]; then
+              echo "Skipping document "$i""
+              continue
+            elif [[ "${i}" =~ .*(internal|proto)/.* ]] || [[ "${i}" =~ go\.(mod|sum) ]]; then
               arr=(${SERVICES[@]})
               echo "All services needs to be built"
```

Comment:
while we're at it, the following could be fixed, too:
* typo/grammar: should be "All services *need* to be built.".
* also L41, should instead read "Check the repository for changes".
* also L24 of `CD-pipeline-dev.yml`, should instead read "Check the PR for changes since the latest commit.".